mlflow/models/evaluation/base.py

```diff
@@ -1278,20 +1299,25 @@ def evaluate(
           specified, unless the ``evaluator_config`` option **log_model_explainability** is
           explicitly set to ``True``.
 
-    :param model: A pyfunc model instance, or a URI referring to such a model.
+    :param model: One of the following:
+                  - A pyfunc model instance
+                  - A URI referring to a pyfunc model
+                  - A function
+                  - A string name of a column from ``data`` that contains model outputs
 
     :param data: One of the following:
 
                  - A numpy array or list of evaluation features, excluding labels.
 
-                 - A Pandas DataFrame or Spark DataFrame, containing evaluation features and
-                   labels. If ``feature_names`` argument not specified, all columns are regarded
-                   as feature columns. Otherwise, only column names present in ``feature_names``
-                   are regarded as feature columns. If it is Spark DataFrame, only the first 10000
-                   rows in the Spark DataFrame will be used as evaluation data.
+                 - A Pandas DataFrame or Spark DataFrame, containing evaluation features,
+                   labels, and optionally model outputs. If ``feature_names`` argument not
+                   specified, all columns are regarded as feature columns. Otherwise, only
+                   column names present in ``feature_names`` are regarded as feature columns.
+                   If it is Spark DataFrame, only the first 10000 rows in the Spark DataFrame
+                   will be used as evaluation data.
 
-                 - A :py:class`mlflow.data.dataset.Dataset` instance containing evaluation features
-                   and labels.
+                 - A :py:class`mlflow.data.dataset.Dataset` instance containing evaluation
+                   features, labels, and optionally model outputs.
```

Comment:
The mlflow dataset does not have a concept of model output. It only has one or more feature columns and one label column.
We need to add one or more model_output columns to it. We need to ask users to specify the model output keys in evaluator_config.

For function inputs, we also need to ask users to specify the model output keys in evaluator_config.